packages/vscode-extension/src/project/deviceSession.ts

```diff
@@ -116,6 +122,10 @@ export class DeviceSession implements Disposable {
   private async bootDevice(deviceSettings: DeviceSettings) {
     this.eventDelegate.onStateChange(StartupMessage.BootingDevice);
     await this.device.bootDevice();
+    if (await this.device.checkLocale(deviceSettings.locale)) {
+      await this.device.changeLocale(deviceSettings.locale);
+      await this.device.resetDevice();
+    }
```

Comment:
seems like this is implementation detail for a given device, shouldn't it be a part of changeSetting implementation that's specific to device rather than being handled in a generic way?